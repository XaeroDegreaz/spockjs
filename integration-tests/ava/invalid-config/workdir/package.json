{
  "ava": {
    "files": [
      "ava.js"
    ],
    "failWithoutAssertions": false,
    "babel": {
      "testOptions": {
        "plugins": [
          [
            "../../../../packages/babel-plugin-spock/src/index.ts",
            {
              "asdf": true
            }
          ]
        ]
      }
    }
  },
  "babel": {
    "presets": [
      [
        "@babel/preset-env",
        {
          "targets": {
            "node": 6
          }
        }
      ]
    ]
  }
}
