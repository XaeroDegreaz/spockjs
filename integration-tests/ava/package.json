{
  "ava": {
    "files": ["ava.js"],
    "failWithoutAssertions": false,
    "babel": {
      "testOptions": {
        "plugins": ["../../packages/babel-plugin-spock/src/index.ts"]
      }
    }
  },
  "babel": {
    "presets": [["@babel/preset-env", { "targets": { "node": 6 } }]]
  }
}
