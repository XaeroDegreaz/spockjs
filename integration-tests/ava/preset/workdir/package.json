{
  "ava": {
    "files": [
      "ava.js"
    ],
    "babel": {
      "testOptions": {
        "plugins": [
          [
            "../../../../packages/babel-plugin-spock/src/index.ts",
            {
              "presets": [
                "@spockjs/preset-runner-ava"
              ]
            }
          ]
        ]
      }
    }
  },
  "babel": {
    "presets": [
      [
        "@babel/preset-env",
        {
          "targets": {
            "node": 6
          }
        }
      ]
    ]
  }
}
