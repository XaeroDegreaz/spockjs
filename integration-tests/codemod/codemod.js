expect: 1 === 1;

expect: {
  2 === 2;
  3 === 4;
}
