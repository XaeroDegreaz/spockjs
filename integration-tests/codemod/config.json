{
  "plugins": [
    [
      "@spockjs/babel-plugin-spock",
      {
        "powerAssert": false,
        "autoImport": "assert",
        "assertFunctionName": "assert"
      }
    ]
  ]
}
