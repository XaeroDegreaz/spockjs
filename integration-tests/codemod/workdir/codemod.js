import * as sinon from 'sinon';
import { env } from 'process';

const { suite } = env;

if (suite === 'assertion') {
  expect: 1 === 1;

  expect: {
    2 === 2;
    3 === 4;
  }
}

if (suite === 'interaction') {
  const obj = { method: () => {} };
  const m = sinon.mock(obj);

  mock: 2 * m.method();

  obj.method();

  verify: m;
}
