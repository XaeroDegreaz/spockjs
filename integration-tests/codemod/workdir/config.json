{
  "plugins": [
    [
      "@spockjs/babel-plugin-spock",
      {
        "powerAssert": false,
        "autoImport": "assert",
        "assertFunctionName": "assert",
        "presets": ["@spockjs/preset-sinon-mocks"]
      }
    ]
  ]
}
