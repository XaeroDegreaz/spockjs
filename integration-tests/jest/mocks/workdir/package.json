{
  "babel": {
    "presets": [
      [
        "@babel/preset-env",
        {
          "targets": {
            "node": 6
          }
        }
      ]
    ],
    "plugins": [
      [
        "@spockjs/babel-plugin-spock",
        {
          "presets": [
            "@spockjs/preset-runner-jest",
            "@spockjs/preset-jest-mocks"
          ]
        }
      ]
    ]
  },
  "jest": {
    "testEnvironment": "node",
    "testRegex": "jest.js",
    "transform": {
      "^.+\\.js$": "babel-jest",
      "^.+packages.+src.+\\.ts$": "ts-jest"
    }
  }
}
