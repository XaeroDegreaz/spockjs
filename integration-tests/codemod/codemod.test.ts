import { sync as run } from 'cross-spawn';
import { readFileSync as read, unlinkSync as del } from 'fs';
import { resolve } from 'path';

import {
  modulePath,
  requireBabelJitArgs,
  runWithTypescriptJit,
} from '../utils';

// mark implicit dependencies for jest
() =>
  require('./workdir/codemod.js') &&
  require('./workdir/config.json') &&
  require('./workdir/package.json');

const babelCli = resolve(modulePath('@babel/cli'), 'bin', 'babel');
const cwd = resolve(__dirname, 'workdir');
const babelConfig = resolve(cwd, 'config.json');
const inFile = resolve(cwd, 'codemod.js');
const outFile = resolve(cwd, 'codemod.out.js');

beforeAll(() => {
  const { status } = runWithTypescriptJit(
    [
      babelCli,
      '--no-babelrc',
      '--config-file',
      babelConfig,
      '-o',
      outFile,
      inFile,
    ],
    { cwd },
  );
  if (status) throw 'babel codemod exited with status ' + status;
});

afterAll(() => del(outFile));

test('generates code that looks hand-written', () => {
  expect(read(outFile).toString()).toMatchSnapshot();
});

test('generates code that performs the assertion', () => {
  const { status, stderr } = run('node', [...requireBabelJitArgs, outFile], {
    cwd,
  });

  expect: status === 1;
  expect(stderr.toString()).toMatch(
    /AssertionError \[ERR_ASSERTION\]: \(3 === 4\) is not truthy/,
  );
});
