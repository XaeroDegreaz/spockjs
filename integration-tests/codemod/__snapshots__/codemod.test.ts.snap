// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generates code that looks hand-written 1`] = `
"import assert from \\"assert\\";
assert(1 === 1);
assert(2 === 2);
assert(3 === 4);
"
`;
